lib/src/base-styles/_nx-alert.scss

```diff
@@ -5,38 +5,57 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 @use '../scss-shared/nx-container-helpers';
+@use '../scss-shared/nx-dark-mode-helpers';
 
 /*
   #Alerts
 */
-@mixin close-btn-interaction-state($color, $hover-value: 80, $active-value: 90) {
-  &:hover {
-    background-color: var(--nx-swatch-#{$color}-#{$hover-value});
-  }
-  &:active {
-    background-color: var(--nx-swatch-#{$color}-#{$active-value});
-  }
-}
 
 .nx-alert {
   @include nx-container-helpers.container-horizontal;
 
   align-items: first baseline;
-  background-color: var(--nx-swatch-white);
-  border: 1px solid var(--nx-swatch-grey-05);
+  background-color: var(--nx-color-component-background);
+  border: 1px solid var(--nx-color-text-stark);
   border-radius: var(--nx-border-radius);
   box-sizing: border-box;
+  color: var(--nx-color-text);
   display: flex;
   margin: 16px 0 24px 0;
   padding: var(--nx-spacing-4x);
   text-align: initial;
   max-width: var(--nx-width-text-wrapping);
 
   @each $type in warning, error, info, success {
-    &--#{$type} > .nx-icon {
-      color: var(--nx-color-alert-#{$type});
+    &--#{$type} {
+      border-color: var(--nx-color-alert-#{$type}-border);
+      background-color: var(--nx-color-alert-#{$type}-background);
+      color: var(--nx-color-alert-text);
+
+      > .nx-icon {
+        color: var(--nx-color-alert-#{$type}-icon);
+      }
+
+      > .nx-btn--close {
+        color: var(--nx-color-alert-close-btn);
+
+        &:hover {
+          background-color: var(--nx-color-alert-#{$type}-close-btn-hover);
+          color: var(--nx-color-alert-close-btn);
+        }
+
+        &:focus:not(:active) {
+          outline-color: var(--nx-color-alert-close-btn-focus);
+        }
+
+        &:active {
+          background-color: var(--nx-color-alert-#{$type}-close-btn-active);
+          color: var(--nx-color-alert-close-btn);
```

Comment:
Most of the `color` properties here use variable names that aren't different in each iteration of the loop.  Can those be pulled out of the loop?

Reply:
> Most of the `color` properties here use variable names that aren't different in each iteration of the loop. Can those be pulled out of the loop?

Unfortunately not. The hover & active styling on icon-only buttons overrides `--nx-color-alert-close-btn` in both the active and hover states, so it needs to be explicitly re-defined.